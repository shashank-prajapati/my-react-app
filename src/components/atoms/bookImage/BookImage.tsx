import React from "react";

type Props = {
  source: string;
  width?: string | number | undefined
};

function BookImage(props: Props) {
  const { source,width } = props;
  return (
    <img
      style={{ display: "block" }}
      src={source}
      width={width}
      alt="Book Cover"
    ></img>
  );
}

export default BookImage;
