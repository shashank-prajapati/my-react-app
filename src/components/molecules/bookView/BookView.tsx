import { Box,Typography} from "@mui/material";
import React from "react";
import TimeLeft from "../timeLeft/TimeLeft";
import ArrowForwardIcon from "@mui/icons-material/ArrowForward";
import Image from "../../atoms/image/Image";
import ButtonAtom from "../../atoms/button/ButtonAtom";

export interface BookProps {
  bookTitle: string;
  bookDesc: string;
  bookAuthor: string;
  time: string;
  coverImage: string;
  onClick?: any;
}

const BookView = (props: BookProps) => {
  const { bookTitle, bookDesc, bookAuthor, time, coverImage, onClick } = props;
  return (
    <Box display="flex" flexDirection="row" justifyContent="space-between">
      <Box flexDirection="column">
        <Box pb={6}>
          <Typography variant="heading" color="navTextColors.main">
            {bookTitle}
          </Typography>
        </Box>
        <Box pb={6}>
          <Typography variant="subtitle3" color="navTextColors.main">
            {bookDesc}
          </Typography>
        </Box>
        <Box pb={4.5}>
          <Typography variant="body1" color="extraTextColor.main">
            {bookAuthor}
          </Typography>
        </Box>
        <Box pb={20.5}>
          <TimeLeft time={time}></TimeLeft>
        </Box>
        <Box flexDirection="row" justifyContent="space-between" width="100%">
          <ButtonAtom
            variant="outlined"
            textColor="primary.main"
            sx={{
              borderColor: "navTextColors.main",
              mr: 6,
            }}
          >
            Read Now
          </ButtonAtom>
          <ButtonAtom
            variant="contained"
            textColor="navTextColors.main"
            backgroundColor="primary.main"
            sx={{
              mr: 6,
            }}
            onClick={() => onClick()}
          >
            Finished Reading
          </ButtonAtom>
          <ButtonAtom
            sx={{
              mr: 6,
            }}
            variant="text"
            textColor="extraTextColor.main"
            endIcon={<ArrowForwardIcon sx={{ color: "extraTextColor.main" }} />}
          >
            Send To Kindle
          </ButtonAtom>
        </Box>
      </Box>
      <Box>
        <Box>
          <Image
            source={`${process.env.PUBLIC_URL}/assets/cover_image/${coverImage}`}
            imgWidth="302px"
            imgHeight="302px"
          />
        </Box>
      </Box>
    </Box>
  );
};

export default BookView;
