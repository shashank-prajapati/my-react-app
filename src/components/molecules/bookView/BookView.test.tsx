import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import BookView from "./BookView";
import { Typography } from "@mui/material";
import BookImage from "../../../media/coverImage.png";

describe("Testing Book View Molecule", () => {
  it("should match the Book View snapshot", () => {
    const { container } = render(
      <BookView
        time="13"
        bookAuthor={"Jim Collins & Bill Lazier"}
        bookTitle="Beyond Entrepreneurship"
        bookDesc="Turning Your Business into an Enduring Great Company"
        coverImage="Entrepreneurship.png"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should call on click method of button", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <BookView
        time="13"
        bookAuthor={"Jim Collins & Bill Lazier"}
        bookTitle="Beyond Entrepreneurship"
        bookDesc="Turning Your Business into an Enduring Great Company"
        coverImage="Entrepreneurship.png"
        onClick={handleClick}
      />
    );
    fireEvent.click(screen.getByText("Finished Reading"));
    expect(handleClick).toHaveBeenCalled();
  });
});
