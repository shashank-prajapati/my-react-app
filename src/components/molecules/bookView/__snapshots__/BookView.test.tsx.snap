// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Book View Molecule should match the Book View snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-9jay18"
  >
    <div
      class="MuiBox-root css-cgq59l"
    >
      <div
        class="MuiBox-root css-1guwpmc"
      >
        <span
          class="MuiTypography-root MuiTypography-heading css-g6ah8q-MuiTypography-root"
        >
          Beyond Entrepreneurship
        </span>
      </div>
      <div
        class="MuiBox-root css-1guwpmc"
      >
        <span
          class="MuiTypography-root MuiTypography-subtitle3 css-g6ah8q-MuiTypography-root"
        >
          Turning Your Business into an Enduring Great Company
        </span>
      </div>
      <div
        class="MuiBox-root css-17gg857"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-zrh9k0-MuiTypography-root"
        >
          Jim Collins & Bill Lazier
        </p>
      </div>
      <div
        class="MuiBox-root css-1dnsdzb"
      >
        <div>
          <div
            class="MuiBox-root css-70qvj9"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
              focusable="false"
            >
              clockIcon.svg
            </svg>
            <span
              class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
            >
              13
              -minute read
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-2iqtls"
      >
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-hpr9xy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Read Now
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-7aavsb-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Finished Reading
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-mghlvy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Send To Kindle
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1f67d23-MuiSvgIcon-root"
              data-testid="ArrowForwardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-0"
      >
        <img
          alt="Book"
          height="302px"
          src="/assets/cover_image/Entrepreneurship.png"
          style="display: block;"
          width="302px"
        />
      </div>
    </div>
  </div>
</div>
`;
